networkmanager/src/main/java/com/dezen/riccardo/networkmanager/NetworkDictionary.java

```diff
@@ -0,0 +1,318 @@
+     * @param peer the Peer to find
+     * @return true if it exists, false otherwise.
+     */
+    public boolean contains(SMSPeer peer){
+        if(peer == null) return false;
+        return peers.containsKey(peer.getAddress());
+    }
+
+    /**
+     * Adds a new resource. Null or invalid Resource won't be added.
+     * @param newResource the new Resource, whose key does not already exist
+     */
+    @Override
+    public boolean addResource(StringResource newResource) {
+        if(newResource == null || !newResource.isValid() || contains(newResource)) return false;
+        resources.put(newResource.getName(), newResource.getValue());
+        return true;
+    }
+
+    /**
+     * Removes the Resource with the matching key, if it exists.
+     * Null or invalid Resource won't be searched for.
+     * @param resourceToRemove the Resource to remove
+     * @return true if it was removed, false otherwise
+     */
+    @Override
+    public boolean removeResource(StringResource resourceToRemove) {
+        if(resourceToRemove == null || !resourceToRemove.isValid() || !contains(resourceToRemove))
+            return false;
+        resources.remove(resourceToRemove.getName());
+        return true;
+    }
+
+    /**
+     * Updates the value of a Resource. Null or invalid Resource won't be updated.
+     * @param updatedResource the new value for the Resource
+     * @return true if the resource was found and updated, false otherwise.
+     */
+    @Override
+    public boolean updateResource(StringResource updatedResource) {
+        if(updatedResource == null || !updatedResource.isValid() || !contains(updatedResource))
+            return false;
+        resources.remove(updatedResource.getName());
+        resources.put(updatedResource.getName(), updatedResource.getName());
+        return true;
+    }
+
+    /**
+     * Returns a copy of the list of all Resources. Resources are not copied singularly.
+     * @return a list containing all Resources
+     */
+    @Override
+    public StringResource[] getResources() {
+        StringResource[] resourcesArray = new StringResource[resources.size()];
+        int i = 0;
+        for(Map.Entry<String, String> resourceEntry : resources.entrySet())
+        {
+            resourcesArray[i++] = new StringResource(resourceEntry.getKey(), resourceEntry.getValue());
+        }
+
+        return resourcesArray;
+    }
+
+    /**
+     * Method to tell whether the given Resource exists in this Vocabulary
+     * @param resource the Resource to find
+     * @return true if resource exists, false otherwise.
+     */
+    public boolean contains(StringResource resource){
+        if(resource == null) return false;
+        return resources.containsKey(resource.getName());
+    }
+
+    /**
+     * Class to interact with Peer Database e Resource Database
+     * @author Giorgia Bortoletti
+     */
+    private class NetworkDictionaryDatabase{
+
```

Comment:
Methods in this class should return something to be sure if everything worked fine

Reply:
Ok thanks, I'll improve it